wyvern/experimentation/providers/base.py

```diff
@@ -0,0 +1,53 @@
+# -*- coding: utf-8 -*-
+from abc import ABC, abstractmethod
+from enum import Enum
+from typing import Optional
+
+
+class ExperimentationProvider(str, Enum):
+    EPPO = "eppo"
+
+
+class BaseExperimentationProvider(ABC):
+    """
+    A base class for experimentation providers.
+    All providers should inherit from this and implement the necessary methods.
+    """
+
+    @abstractmethod
+    def get_result(self, experiment_id: str, entity_id: str, **kwargs) -> str:
```

Comment:
we discussed this in zoom today:
I think we should provide `default_value: Optional[Any] = None` as a backup plan for provider service down time for example.

Reply:
[Could support it later] question for us to think about: is the return type always going to be str? shall we also support int/float, boolean, list, dictionary, blabla?

Reply:
I talked to Mathias afterwards in our thread afterwards. They don't need such a functionality so I just excluded it.

https://wyvern-ai.slack.com/archives/C05F8FX8VH6/p1691777068492019

I think the experimentation frameworks just use strings as return types. We can add support if it's needed in the future.

Reply:
no need to do the `default_value` as client can handle the None.